src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigWithFineTunedAlignments.java

```diff
@@ -118,66 +129,103 @@ public final boolean hasOnly2GoodAlignments() {
+        }
     }
 
     public final boolean hasEquallyGoodAlnConfigurations() {
         return hasEquallyGoodAlnConfigurations;
     }
 
+    public final boolean hasIncompletePicture() {
+        if ( hasOnly2GoodAlignments() )
+            return hasIncompletePictureFromTwoAlignments(getHeadAlignment(), getTailAlignment());
+        else
+            return hasIncompletePictureFromMultipleAlignments();
+    }
+
     //==================================================================================================================
 
     /**
-     * An assembly contig with two good alignments indicates to us that
-     * it doesn't have the complete alt haplotype assembled if it shows
+     * This predicate tests if an assembly contig has the full event (i.e. alt haplotype) assembled
+     * Of course, the grand problem of SV is always not getting the big-enough picture
+     * but here we have a more workable definition of what is definitely not big-enough:
+     *
+     * If the assembly contig, with its two (picked) alignments, shows any of the following signature,
+     * it is definitely not giving the whole picture of the alt haplotype,
+     * hence without other types of evidence (or linking breakpoints, which itself needs other evidence anyway),
+     * human-friendly interpretation for them is unreliable.
      * <ul>
      *     <li>
-     *        either alignment's ref span is contained in the other's;
+     *         head and tail alignments contain each other in terms of their reference span,
+     *         regardless if strand switch is involved;
+     *     </li>
+     *     <li>
+     *         if the two alignments map to the same chromosome, and their reference spans overlap, and in the mean time
+     *         <ul>
+     *             <li>
+     *                 there's reference order switch, or
+     *             </li>
+     *             <li>
+     *                 there's strand switch,
+     *         todo: this obsoletes the actual use of SimpleChimera#isCandidateInvertedDuplication()
+     *               and related inverted duplication call code we have now,
+     *               but those could be used to figure out how to annotate which known ref regions are invert duplicated
+     *             </li>
+     *         </ul>
      *     </li>
-     *
-     *    <li>
-     *        if the two alignments map to the same chromosome, and their ref span indicate order switch AND overlap
-     *        (this indicate a duplication, inverted or not, but we don't have the full duplicated region assembled)
-     *    </li>
      * </ul>
-     *
      */
-    public boolean hasIncompletePictureFromTwoAlignments() {
-        return hasIncompletePictureDueToRefSpanContainment()
-                ||
-                (firstAndLastAlignmentMappedToSameChr() && hasIncompletePictureDueToOverlappingRefOrderSwitch());
-    }
+    public static boolean hasIncompletePictureFromTwoAlignments(final AlignmentInterval headAlignment,
```

Comment:
Do we need to have both methods `hasIncompletePictureFromTwoAlignments` and `hasIncompletePictureFromMultipleAlignments`? Couldn't one method do it?